app/src/main/java/com/example/instagramclone/fragments/main_screen/compose/PostItem.kt

```diff
@@ -0,0 +1,215 @@
+            )
+        }
+    }
+}
+
+@Composable
+fun PostBottomBar(savePost: () -> Unit) {
+    var isPostLiked by remember {
+        mutableStateOf(false)
+    }
+    var isPostSaved by remember {
+        mutableStateOf(false)
+    }
+
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(10.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        Row() {
+            BottomBarIcon(
+                onClick = {
+                    isPostLiked = !isPostLiked
+                },
+                painter = painterResource(
+                    id = if (isPostLiked) {
+                        drawable.ic_heart_64_true
+                    } else {
+                        drawable.ic_heart_64_false
+                    }
+                )
+            )
+            BottomBarIcon(
+                onClick = { },
+                painter = painterResource(id = drawable.ic_comment)
+            )
+
+            BottomBarIcon(
+                onClick = { },
+                painter = painterResource(id = drawable.ic_share)
+            )
+        }
+        BottomBarIcon(
+            onClick = {
+                isPostSaved = !isPostSaved
+                savePost()
+            },
+            painter = painterResource(
+                id = if (isPostSaved) {
+                    drawable.ic_save_true
+                } else {
+                    drawable.ic_save_false
+                }
+            )
+        )
+    }
+}
+
+@Composable
+fun BottomBarIcon(
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+    painter: Painter
+) {
+    Icon(
+        modifier = modifier
+            .padding(horizontal = 8.dp)
+            .clickable(interactionSource = MutableInteractionSource(), indication = null) {
+                onClick()
+            },
+        painter = painter,
+        contentDescription = "BottomBarIcon"
+
+    )
+}
+
+@Composable
+fun PostTopbar(postItem: Post, modifier: Modifier = Modifier) {
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(vertical = 10.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            Box(
+                modifier = Modifier
+                    .wrapContentSize()
+                    .padding(horizontal = 10.dp),
+                contentAlignment = Alignment.Center
+            ) {
+                Image(
+                    modifier = Modifier
+                        .size(35.dp),
+                    painter = painterResource(id = drawable.ic_story_border_rainbow),
+                    contentDescription = "story border"
+                )
+                AsyncImage(
+                    modifier = Modifier
+                        .size(30.dp)
```

Comment:
It's best to use the [4-point grid system](https://medium.com/@aratidube12lns/4-point-grid-system-for-more-consistent-interface-design-efea81dea3f3#:~:text=What%20is%20the%204%2Dpoint,padding%2C%20margins%20and%20line%20heights.) wherever possible.